### PR TITLE
Mesh 534/statistic module

### PR DESCRIPTION
**Pull request by fmiguelgarcia**
New Module `Statistics`:
- `update_transfer_stats` only tracks the number of investors per asset. It is called from `asset::_transfer` function.

Code Hygiene:
  - `Statistics` Unit Test at `src/test/statistics.rs`
  - `Asset` unit test now uses `TestStorage` (it reduces ~230 lines of code) and it has been moved to `src/test/asset.rs`

### CODE REVIEW

File: runtime/src/asset.rs [hunk 1]
```diff
@@ -1612,1586 +1621,3 @@ impl<T: Trait> Module<T> {
         Ok(())
     }
 }
```
Comment (maxsam4) [inline runtime/src/asset.rs]: We need to update the transfer stats on all Issue/mint and redeem functions as well. See all the places where `_update_checkpoint` is being called. Infact, we can maybe merge the status updation and checkpoints updation. I wonder if storing checkpoints in the stats module is a good idea.

File: runtime/src/statistics.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if amount != 0u128.into() {
+            let counter = Self::investor_count_per_asset(ticker);
+            let mut new_counter = counter;
+
+            if updated_from_balance == 0u128.into() {
+                new_counter = new_counter.checked_sub(1).unwrap_or(new_counter);
+            }
+            if updated_to_balance == amount {
+                new_counter = new_counter.checked_add(1).unwrap_or(new_counter);
+            }
```
Comment (maxsam4) [inline runtime/src/statistics.rs]: Need to also check that `amount != 0` because 0 amount transfers are also possible.

Comment (fmiguelgarcia) [reply runtime/src/statistics.rs]: yes, it is done in line 34.

**Comment by fmiguelgarcia** [comment]
> Everything is fine beside what @maxsam4 points out. I also want test unification, It means if we are ripping out the test from the module file itself then it is good to have `{modulename}_mock.rs` as a file name for mocking runtime and the actual test should `{modulename}_test.rs` and the structure that will be used for testing should be named as either `Test` or `TestRuntime`

File: runtime/src/asset.rs (hunk 1 above)
Comment (fmiguelgarcia) [reply runtime/src/asset.rs]: Done. Now updated balance parameters (from & to) are optional to cover _redeem_ and _mint_ operations.

**Pull request merged**